articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+We can use annotation instances as reference target.
+
+Assuming node `X` mentions node `Y` in ``X``'s annotation instances, we say `X` is _annotated by_ `Y`, and `Y` has _annotated node_ `Y`.
+``Y``'s annotated node is always also ``Y``'s parent node.
+
+[plantuml, annotationInstance, svg]
+----
+hide empty members
+
+class X
+annotation Y
+
+X -[#transparent]> Y: < parent
+X -[#transparent]> Y: < annotated node
+X -[#transparent]> Y: > annotated by <U+00a0><U+00a0><U+00a0>
+X *- Y: > <<annotations>>
+----
+
+Annotation instances share the lifecycle of their annotated node; when we delete the annotated node, the annotation instance gets also deleted.
+
+==== Applicable classifiers
+The classifier of an annotation instance must be an annotation.
+Annotations are similar to concepts, and both are classifiers.
+But an annotation specifies which other classifier it _annotates_.
+The instance of that annotation must adhere to this constraint.
+
+As example, assume we have classifiers `Function` and `Argument`.
+We also have an annotation `SourcePosition` that `annotates = Function`.
+On M1, we have node `1` of `Function`, and node `3` of `Argument`.
+With node `34`, we have an annotation instance of `SourcePosition`.
+Node `1:Function` can be _annotated by_ node `34:SourcePosition`, because `SourcePosition` states it _annotates_ `Function`.
+In contrast, node `3:Argument` cannot be _annotated by_ node `35:SourcePosition`, because `Argument` is not compatible with `Function` (the kind of classifier ``SourcePosition`` _annotates_).
+
+[plantuml,annotation,svg]
+----
+hide empty members
+
+package M2 <<Rectangle>> {
+  class Function <<Classifier>> {
+    name: String
+  }
+
+  class Argument <<Classifier>> {
+  }
+  Function *- "0..*" Argument: parameters
+
+  annotation SourcePosition <<Annotation>> {
+    line: Integer
+    column: Integer
+  }
+  SourcePosition .> Function: annotates
+}
+
+package M1 <<Rectangle>> {
+  class f1 as "1:Function" <<Node>> {
+    name = "doIt"
+  }
+
+  class a1 as "3:Argument" <<Node>> {
+    name = "action"
+  }
+  f1 *- a1: parameters
+
+  annotation an1 as "34:SourcePosition" <<Node>> {
+    line = 2
+    column = 13
+  }
+  an1 -* f1: <<annotations>>
+
+  annotation an2 as "35:SourcePosition" <<Node>>  #line:red {
+    line = 2
+    column = 18
+  }
+  a1 *- an2: <<annotations>>
+}
+
+M2 -[hidden]- M1
+
+'Function <.. f1: <<classifier>>
+'Argument <.. a1: <<classifier>>
+'SourcePosition <.. an1: <<classifier>>
+'SourcePosition <. an2: <<classifier>>
+
+----
+
+An annotation that is applicable to all concepts says it annotates `Node` from LionWeb's built-in library.
+Annotations can _annotate_ other annotations; however, this approach introduces complexities and should be used with great caution.
+
+#TODO:# Do we allow unresolvable annotations?
+
+=== Differences between containment and annotation instance
+Both a contained node `C` and an annotated node `A` share the lifecycle of their parent node `P`.
```

Comment:
Not am annotated node, but an annotation instance, right?

Reply:
technically, "contained" is equivalent to "annotated node" (as established above). Not sure which one is easier to understand here.

Reply:
In the picture in section 2.9: annotated node is `X`, right? But `X` is not contained, it would be a parent. How then "contained" is equivalent to "annotated node"? I am confused now.